src/node/coinstats.cpp

```diff
@@ -206,17 +202,52 @@ static bool GetUTXOStats(CCoinsView* view, BlockManager& blockman, CCoinsStats&
     stats.hashSerialized = hasher.Finalize();
 
     stats.nDiskSize = view->EstimateSize();
+
     return true;
 }
 
-std::optional<CCoinsStats> GetUTXOStats(CCoinsView* view, BlockManager& blockman, CoinStatsHashType hash_type, const std::function<void()>& interruption_point, const CBlockIndex* pindex, bool index_requested)
+std::optional<CCoinsStats> GetUTXOStatsWithHasher(UTXOHasher& hasher, CCoinsView* view, BlockManager& blockman, const std::function<void()>& interruption_point)
 {
-    CCoinsStats stats{};
-    auto hasher = MakeUTXOHasher(hash_type);
+    CBlockIndex* pindex = WITH_LOCK(cs_main, return blockman.LookupBlockIndex(view->GetBestBlock()));
+    CCoinsStats stats = MakeCoinStatsPrefilledWithBlockIndexInfo(pindex);
+
+    GetUTXOStatsWithHasher(hasher, stats, view, blockman, interruption_point);
 
-    if (!GetUTXOStats(view, blockman, stats, *hasher, interruption_point, pindex, hash_type, index_requested)) {
+    return stats;
+}
+
+std::optional<CCoinsStats> GetUTXOStatsWithIndex(CoinStatsIndex& coin_stats_index, const CBlockIndex* pindex)
+{
+    CCoinsStats stats = MakeCoinStatsPrefilledWithBlockIndexInfo(pindex);
+
+    stats.index_used = true;
```

Comment:
Same comment as with `hashSerialized`. `index_used` does not seem to be a value internal to `CCoinsStats` anymore.

Reply:
Although I’ve moved some ***in***-param members out of `CCoinStats`, I think it’s outside the scope of this PR to move ***out***-param members out of `CCoinStats` but would be a worthwhile followup. This is because a proper cleanup of `CCoinStats`' out-params would likely result in a complete split of the struct itself.

Observe that:

- The hashing UTXO stats code path only ever sets the CCoinsStats members:
    - CCoinsStats::CCoinsStats
        - nHeight
        - hashBlock
    - ApplyStats
        - nTransactions
        - nTransactionOutputs
        - total_amount
        - nBogoSize
    - ComputeUTXOStatsWithHasher
        - coins_count
        - hashSerialized
        - nDiskSize
- The indexing UTXO stats code path only ever sets the CCoinsStats members:
    - CCoinsStats::CCoinsStats
        - nHeight
        - hashBlock
    - CoinstatsIndex::LookupStats
        - index_used
        - hashSerialized
        - nTransactionOutputs
        - nBogoSize
        - total_amount
        - total_subsidy
        - total_unspendable_amount
        - total_prevout_spent_amount
        - total_new_outputs_ex_coinbase_amount
        - total_coinbase_amount
        - total_unspendables_genesis_block
        - total_unspendables_bip30
        - total_unspendables_scripts
        - total_unspendables_unclaimed_rewards
- The only 6 common fields that are set:
    - nHeight
    - hashBlock
    - nTransactionOutputs
    - total_amount
    - nBogoSize
    - hashSerialized

So a proper cleanup would make it so that `CCoinStats` only contains the 6 common fields, and have other structs to contain the fields unique to each codepath.